lib/generic-request.js

```diff
@@ -0,0 +1,39 @@
+const isCorrectHttpMethod = require('./check-http-method');
+const HTTP_STATUS_CODES = require('./http-status-codes');
+const rxjs = require('rxjs');
```

Comment:
Maybe it's better to require the methods you'll use in your code? There is no need in whole package in your case and I think that that could reduce the final package size for dependent projects.

Reply:
Make sense to require only the things we really need, but how does that changes an overall package size since anyway rxjs is living inside HttpRx's node_modules.

Reply:
I've applied destructuring assignment for referencing `Observable` and `throwError`.
Yet would love to hear you thoughts regarding package size reducing since we don't have any build procedure in place, which involves three-shaking.